tests/backward_compatibility_tests.sh

```diff
@@ -38,6 +38,8 @@ pip uninstall -y skypilot
 pip install uv
 uv pip install --prerelease=allow "azure-cli>=2.65.0"
 uv pip install -e ".[all]"
+# In case there are some clusters left from previous tests
+sky down ${CLUSTER_NAME}* -y
```

Comment:
If we have `sky down` all the cluster in the end of this script, why should we clean up again here? We should make sure the script is self-contained, instead of across multiple runs.

Reply:
If it exits in the middle of the script, the script will not terminate the clusters because the termination codes are at the end of the script.

Reply:
I will use `trap` to terminate it.